### DISCUSSION TITLE
New Accounts feature

### DISCUSSION THREAD

**Discussion by samejr** [apihero-run/jsonhero-web#56] [Ideas]
# Hello! 👋 Let's talk Accounts!

I'd like to start a discussion about a new Accounts feature for JSON Hero that I've been wireframing. I'd love to hear your thoughts and get as much feedback and suggestions as possible so we can make this as awesome as possible. 🚀

# 1. Let's start with signing up

The header would get a few new buttons:

![CleanShot 2022-05-06 at 15 08 21@2x](https://user-images.githubusercontent.com/7555566/167154034-74138c16-352a-4a9d-a1f7-a4a8abed6e95.png)

If you click 'Sign in' there'll be a new Sign in modal. For version 1, we thought it best to add Github account Sign in only. We can add more at a later date.

![CleanShot 2022-05-06 at 15 09 19@2x](https://user-images.githubusercontent.com/7555566/167154722-67e4729c-8229-4a5d-9122-ee32660abc3f.png)

# 2. Next up - a new Dashboard

The main purpose of the dashboard is to keep a history and give you control of the documents you create and the ones you've viewed. I've also combined this page with your profile information. This page would become https://jsonhero.io when logged in.

![CleanShot 2022-05-06 at 14 42 18@2x (1)](https://user-images.githubusercontent.com/7555566/167157392-5be924f5-38fe-4ead-8b51-c9bf1bcfafc0.png)

# 3. How would sharing work?

When you are logged in, any documents you create will remain private to you by default.

![CleanShot 2022-05-06 at 15 01 28@2x](https://user-images.githubusercontent.com/7555566/167158878-f7b76270-ae8d-44d7-aedb-943b4aa67d82.png)

You can create a shareable link for each document by toggling on the Sharing feature in this new Share modal:

![CleanShot 2022-05-06 at 15 12 42@2x](https://user-images.githubusercontent.com/7555566/167155690-3c4ca95a-77d5-47a6-bae0-93fb683e470f.png)

To keep link sharing secure, logged our users would still be able to use JSON Hero normally, but saving and sharing documents would be restricted with a time limit.

![CleanShot 2022-05-06 at 15 16 46@2x](https://user-images.githubusercontent.com/7555566/167159639-cf2dd288-7e4c-4fc4-aad3-3471332ee9fa.png)

# 4. Anything else?

Just a few more things. To keep the UI consistent, we're proposing to give the New document panel some more real estate by putting it into a modal:

![CleanShot 2022-05-06 at 15 07 22@2x](https://user-images.githubusercontent.com/7555566/167155711-306dbe3a-fd73-4f20-b7d3-65ac8a0c03a0.png)

We also want a quick way to access some of the meta data in JSON documents so are proposing a new slide over panel accesses from a new toggle icon in the side menu bar:  

![CleanShot 2022-05-06 at 15 21 59@2x](https://user-images.githubusercontent.com/7555566/167155948-584101dd-74ec-42d4-843c-78a3d4cccfbd.png)

The Accounts dropdown menu would link to the dashboard, display your top 5 starred documents and log you out. Oh, and we're proposing to add a new toast feature for better notifications:

![CleanShot 2022-05-06 at 15 19 12@2x](https://user-images.githubusercontent.com/7555566/167156075-013abe3f-bfad-4b18-bc35-6126c82fcb9e.png)

What do ya think? Please drop a comment or ask questions below.

**Comment by matt-aitken** [comment]
I want to double check what happens if I make a file public and then change it back to private. Presumably the URL will return a 404 for other people looking at it?

**Reply by om-ray** [reply]
that would make sense, something like what github does with their 404 pages would be nice

**Reply by samejr** [reply]
Yes it would return a 404 - we have that already: https://jsonhero.io/j/d9udW60cLOok1

**Reply by om-ray** [reply]
i think that 404 page fits great with the overall theme

**Comment by matt-aitken** [comment]
I think the top-bar is getting a bit crowded on the right hand side.


A few of these operations feel like they're related to the current file. Can share and fork move to be near the title in the middle?

**Reply by om-ray** [reply]
instead of having words do you think it would be possible to use icons? I feel like it would help with the litter and also make it a lot easier to understand at a glance instead of having to read

**Reply by samejr** [reply]
Focusing on icons to simplify it is a good idea. If you'd like to get involved and take a swing on some ideas, that would be awesome 🥇

**Reply by samejr** [reply]
In case it's useful, here are some things I'll be considering when coming up with solutions for this design:

- Refer to other products that solve the clutter problem well. Info-dense products like Airtable, Asana, CodeSandbox etc might be good candidates to reference.
- The hierarchy for the links will likely change if you're logged in vs logged out. 
- Per @matt-aitken's point above, a nice way to split the info might be to collect the file-specific links together somehow.
- We might want to retain a strong promotion CTA – like the Github Stars button so might be worth not demoting everything in the hierarchy too far. 

Food for thought…

**Reply by om-ray** [reply]
I was thinking that if the person isn't signed in we could put the file-related options in some sort of collapsible drop-down menu so that things like forking, starring, and signing up would be the main focus.

This would change after they sign up, with forking and starring as simple icons and the file-related links as buttons.

**Comment by matt-aitken** [comment]
I don't 100% understand whether sharing is available or not when logged out? Is the suggestion that you can share but the file would expire after 24 hours?

**Reply by matt-aitken** [reply]
Proposal
- Logging in means you can create long-lived documents
- Sharing is always available, logged in or not
- If sharing a document that is going to expire we make it clear it's going to expire. Give the option to login to make the document permanent.

**Reply by samejr** [reply]
I like this suggestion 👍
